dotcom-rendering/src/web/components/KeyEventsCarousel.importable.tsx

```diff
@@ -0,0 +1,204 @@
+`;
+
+const buttonContainerStyles = css`
+	display: flex;
+	height: 34px;
+	margin-top: 10px;
+	margin-bottom: ${space[5]}px;
+	position: absolute;
+	bottom: 0;
+	justify-content: space-between;
+	width: 100%;
+	padding-right: 40px;
+`;
+
+const buttonStyles = css`
+	border: 0 none;
+	border-radius: 100%;
+	height: 34px;
+	width: 34px;
+	cursor: pointer;
+	padding: 0;
+	background-color: ${neutral[0]};
+	bottom: 4px;
+	&:active,
+	&:hover {
+		outline: none;
+		background-color: ${brandAlt[400]};
+		svg {
+			fill: ${neutral[7]};
+		}
+	}
+
+	&:focus {
+		outline: none;
+	}
+
+	svg {
+		fill: ${neutral[100]};
+		height: 34px;
+	}
+`;
+
+const disabledButtonStyles = css`
+	background-color: ${neutral[60]};
+	cursor: default;
+
+	&:hover,
+	&:focus {
+		background-color: ${neutral[60]};
+
+		svg {
+			fill: ${neutral[100]};
+		}
+	}
+`;
+
+export const KeyEventsCarousel = ({
+	keyEvents,
+	filterKeyEvents,
+	format,
+}: Props) => {
+	const [activeIndex, setActiveIndex] = useState(0);
+	const [carousel, setCarousel] = useState<HTMLElement>();
+	const [cards, setCards] = useState<Element[]>();
+	const cardWidth = 200;
+
+	// we only want to count cards fully in view so we round down.
+	const cardsOnScreen = carousel
+		? Math.floor(carousel.clientWidth / cardWidth)
+		: 0;
+
+	const isFirstCard = activeIndex === cardsOnScreen;
+	const isLastCard = activeIndex === keyEvents.length - 1;
+
+	useEffect(() => {
+		const carouselElement = document.getElementById('key-event-carousel');
+		const cardsArray = Array.from(
+			document.querySelectorAll('#key-event-card'),
+		);
+		if (carouselElement) setCarousel(carouselElement);
+		if (cardsArray) setCards(cardsArray);
+	}, []);
+
+	useEffect(() => {
+		const handleIntersect = (entries: IntersectionObserverEntry[]) => {
+			const entry = entries.find((e) => e.isIntersecting);
+			const index = cards?.findIndex((e) => e === entry?.target);
+			if (index && index !== -1) setActiveIndex(index);
+		};
+
+		const observer = new window.IntersectionObserver(handleIntersect, {
+			root: carousel,
+			rootMargin: '0% 0% 0% -100%',
+		});
+
+		cards?.forEach((el) => {
+			observer.observe(el);
+		});
+
+		return () => observer.disconnect();
+	}, [activeIndex, cards, carousel]);
+
+	const goPrevious = () => {
+		if (carousel && !isFirstCard) carousel.scrollLeft -= cardWidth;
+	};
+
+	const goNext = () => {
+		if (carousel && !isLastCard) carousel.scrollLeft += cardWidth;
+	};
+
+	const transformedKeyEvents = keyEvents
+		.filter((keyEvent) => {
+			return keyEvent.title && keyEvent.blockFirstPublished;
+		})
+		.map((keyEvent) => {
+			return {
+				text: keyEvent.title || '', // We fallback to '' here purely to keep ts happy
+				url: `?filterKeyEvents=${filterKeyEvents}&page=with:block-${keyEvent.id}#block-${keyEvent.id}`,
+				date: new Date(keyEvent.blockFirstPublished || ''), // We fallback to '' here purely to keep ts happy
+			};
+		});
+	return (
+		<div id="key-event-carousel" css={carouselStyles}>
```

Comment:
This isn't possible as far as I can tell, I have tried a few variations but we need two here so that one controls the hidden overflow of blocks and the other allows for overflow scrolling